### PR TITLE
Android Cipher sample added

### PR DESCRIPTION
**Pull request by fullmetaltac**
Android Cipher sample added

### CODE REVIEW

**Comment by pjcollins** [comment]
@fullmetaltac could we include a repository wide .gitignore update in this PR to make sure the `.vs` artifacts are not included? This could be added to similar artifacts around here:
https://github.com/xamarin/monodroid-samples/blob/master/.gitignore#L6

File: android-o/AndroidCipher/AndroidCipher/AndroidCipher.csproj [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+    <OutputType>Library</OutputType>
+    <AppDesignerFolder>Properties</AppDesignerFolder>
+    <RootNamespace>AndroidCipher</RootNamespace>
+    <AssemblyName>AndroidCipher</AssemblyName>
+    <FileAlignment>512</FileAlignment>
+    <AndroidApplication>True</AndroidApplication>
+    <AndroidResgenFile>Resources\Resource.Designer.cs</AndroidResgenFile>
+    <AndroidResgenClass>Resource</AndroidResgenClass>
+    <GenerateSerializationAssemblies>Off</GenerateSerializationAssemblies>
+    <AndroidUseLatestPlatformSdk>True</AndroidUseLatestPlatformSdk>
```
Comment (pjcollins) [inline android-o/AndroidCipher/AndroidCipher/AndroidCipher.csproj]: Let's disable the Use latest platform SDK flag, as this is intended to run against v8.1 (or higher) and we don't yet install that by default.

File: android-o/AndroidCipher/AndroidCipher/Constants.cs [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+{
+    public class Constants
+    {
+        public static readonly int AesKeySize = 128;
+        public static readonly int GcmTagLength = 16;
+        public static readonly int GcmNonceLength = 12;
+
+        public static readonly string Charset = "UTF-8";
+        public static readonly string Algorithm = "AES";
+        public static readonly string Transformation = "AES/GCM/NoPadding";
```
Comment (pjcollins) [inline android-o/AndroidCipher/AndroidCipher/Constants.cs]: Do you think it makes sense to add a spinner to the app that would allow us to demonstrate multiple encryption algorithms? Looking at the 8.1 API docs it looks like there may be more options we can expose here?

File: android-o/AndroidCipher/README.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
+AndroidCipher
+=====
+
+AndroidCipher is a simple project which encrypts and decrypts string with usage of  android cipher.
+The reason of creating this project is to test new methods in android 8.1.
```
Comment (pjcollins) [inline android-o/AndroidCipher/README.md]: I would maybe reword this line slightly, to instead say something like  "This sample demonstrates some of the new encryption algorithms available in Android 8.1"

File: android-o/AndroidCipher/AndroidCipher/AndroidCipher.csproj (hunk 1 above)
Comment (fullmetaltac) [reply android-o/AndroidCipher/AndroidCipher/AndroidCipher.csproj]: resolved

File: android-o/AndroidCipher/README.md (hunk 3 above)
Comment (fullmetaltac) [reply android-o/AndroidCipher/README.md]: resolved

File: android-o/AndroidCipher/AndroidCipher/Constants.cs (hunk 2 above)
Comment (fullmetaltac) [reply android-o/AndroidCipher/AndroidCipher/Constants.cs]: @pjcollins The main idea is to test that I can retrieve GCMParameters from encryptor to decryptor.
